util/execdetails/execdetails.go

```diff
@@ -52,3 +54,58 @@ func (d ExecDetails) String() string {
 	}
 	return strings.Join(parts, " ")
 }
+
+// RuntimeStats collects executors's execution info.
+type RuntimeStats map[string]*RuntimeStat
+
+// RuntimeStat collects one executor's execution info.
+type RuntimeStat struct {
+	// executor's Next() called times.
+	loop int32
+	// executor consume time.
+	consume int64
+	// executor return row count.
+	rows int64
+}
+
+// NewRuntimeStats creates new executor collector.
+func NewRuntimeStats() RuntimeStats {
+	return RuntimeStats(make(map[string]*RuntimeStat))
+}
+
+// GetRuntimeStat gets execStat for a executor.
+func (e RuntimeStats) GetRuntimeStat(planID string) *RuntimeStat {
+	if e == nil {
+		return nil
+	}
+	runtimecStat, exists := e[planID]
+	if !exists {
+		runtimecStat = &RuntimeStat{}
+		e[planID] = runtimecStat
+	}
+	return runtimecStat
+}
+
+func (e RuntimeStats) String() string {
+	var buff bytes.Buffer
+	buff.WriteString("(")
+	for planID, stat := range e {
+		buff.WriteString(planID + ":" + stat.String() + ",")
+	}
+	buff.WriteString(")")
+	return buff.String()
+}
+
+// Record records executor's execution.
+func (e *RuntimeStat) Record(d time.Duration, rowNum int) {
+	atomic.AddInt32(&e.loop, 1)
+	atomic.AddInt64(&e.consume, int64(d))
+	atomic.AddInt64(&e.rows, int64(rowNum))
+}
+
+func (e *RuntimeStat) String() string {
+	if e == nil {
+		return ""
+	}
+	return fmt.Sprintf("time:%f, loops:%d, rows:%d", time.Duration(e.consume).Seconds()*1e3, e.loop, e.rows)
```

Comment:
it's better to specify the time unit in the string message. for example: `"time:%fms, loops:%d, rows:%d"`